src/utils/multi_transport_connection.ts

```diff
@@ -0,0 +1,75 @@
+import * as t from "io-ts";
+
+import { NonEmptyString } from "italia-ts-commons/lib/strings";
+
+/**
+ * Describes a nodemailer transport connection
+ */
+export const MailTransportConnection = t.interface({
+  password: t.string,
+  transport: NonEmptyString,
+  username: t.string
+});
+
+export type MailTransportConnection = t.TypeOf<typeof MailTransportConnection>;
+
+/**
+ * An array of nodemailer transport connections
+ */
+export const MailMultiTransportConnections = t.readonlyArray(
+  MailTransportConnection
+);
+
+export type MailMultiTransportConnections = t.TypeOf<
+  typeof MailMultiTransportConnections
+>;
+
+/**
+ * Parses a multi email provider connection string.
+ *
+ * Multiple providers can be specified as following:
+ *
+ * transport:username:password[;transport:username:pasword][;transport:username:pasword]...
+ *
+ */
+function parseMultiProviderConnection(
+  conn: string
+): MailMultiTransportConnections {
+  return conn
+    .split(";")
+    .map(providerStr => {
+      const [transport, username, password] = providerStr.split(":");
+      if (
+        NonEmptyString.is(transport) &&
+        username !== undefined && // allow empty username
+        password !== undefined // allow empty password
+      ) {
+        return {
+          password,
+          transport,
+          username
+        };
+      }
+      return undefined;
```

Comment:
nit: if you want to avoid the cast here you can return an option instead of undefined and call catOptions on the resulting array: https://github.com/gcanti/fp-ts/blob/1.17.0/src/Array.ts#L1311